src/libsystemd-network/dhcp-identifier.c

```diff
@@ -168,7 +168,7 @@ static int dhcp_identifier_set_duid_uuid(struct duid *ret_duid, size_t *ret_len)
         unaligned_write_be16(&ret_duid->type, DUID_TYPE_UUID);
         memcpy(ret_duid->raw.data, &machine_id, sizeof(machine_id));
 
-        *ret_len = sizeof(ret_duid->type) + sizeof(machine_id);
+        *ret_len = offsetof(struct duid, uuid) + sizeof(machine_id);
```

Comment:
hmme, the memcpy above writes to `ret_duid->raw.data` but here you reference `struct duid`'s `uuid` field instead. It sould point to the same place, but I guess it would be better to fix the memcpy() to use `ret_duid->uuuid.uuid` instead.